src/ucp/core/ucp_context.c

```diff
@@ -931,17 +935,29 @@ void ucp_context_print_info(ucp_context_h context, FILE *stream)
     fprintf(stream, "#\n");
 }
 
-
 void ucp_context_tag_offload_enable(ucp_context_h context)
 {
+    size_t max_bcopy;
+
+    /* Enable offload, if only one tag offload capable interface is present
+     * (multiple offload ifaces are not supported yet). */
     if (ucs_queue_length(&context->tm.offload_ifaces) == 1) {
-        /* Enable offload, because just one tag offload capable interface is present */
-        context->tm.post_thresh = ucs_max(context->config.ext.tm_thresh,
-                                          sizeof(ucp_request_hdr_t));
+        /* Post threshold for tag offload should not be less than ucp_request_hdr_t
+         * size, because this header may be scattered to user buffer in case of
+         * expected SW RNDV protocol. */
+        context->tm.thresh = ucs_max(context->config.ext.tm_thresh,
+                                     sizeof(ucp_request_hdr_t));
+
+        max_bcopy = ucs_min(context->config.ext.tm_max_bcopy,
+                            context->config.ext.seg_size);
```

Comment:
I don't like this "silent" change in default values. It may have impact on performance. Instead of silently changing the value you have to report an warning and notify user that he used poor configuration.

The same true for tm.thresh